### PR TITLE
Increase timeout 

### PR DESCRIPTION
**Pull request by sirugh**
I just experienced the most brutal hell of a few weeks trying to understand why an environment wouldn't load scripts for the first load of a page. The short answer is that 1 second is not sufficient to load scripts over a bandwidth constricted tunnel. At least not for the first time while the browser makes a connection to whatever server is hosting your script. 

In our case, we would hit the timeout which would try to then [use the loaded script](https://github.com/bazaarvoice/scoutfile/blob/master/lib/browser/loader.js#L98) which wasn't fully loaded yet, resulting in an error. My theory is that even though this results in a timeout, the browser (and intermediate servers) had already cached that resource request, so subsequent page loads were able to load the resource in under 1 second.

In the end, I referenced the [original script](https://github.com/SlexAxton/yepnope.js/blob/master/src/yepnope.js) we sourced for some of this code and found that they used a 10 second timeout, which I feel is adequate.

TL;DR - I wish to up the timeout to a nice 10 seconds. Also split the `script.onerror` callback and the timeout callback so future me has an easier time understanding which callback is at fault.

### CODE REVIEW

**Comment by reason-bv** [comment]
This needs the addition of a section on timeout behavior in the project docs, especially with reference to testing and likely longer load times in some environments, and what to do about that.

It is arguably reasonable to keep the 1 second timeout and to be more aggressive in telling people to put in a larger timeout as needed. Also reasonable to have the longer default timeout. I'm on the fence on that one.

**Comment by taylormck** [comment]
I'm all for lengthening the default timeout.
We have environments where a 1 second timeout is reasonable insufficient: mobile, dial-up, etc.
This module needs to support those environments without forcing all its users to patch it.

**Comment by sirugh** [comment]
>It is arguably reasonable to keep the 1 second timeout and to be more aggressive in telling people to put in a larger timeout as needed. Also reasonable to have the longer default timeout. I'm on the fence on that one.

The loader module does provide a timeout option but discovering that the timeout was the root of the problem was not trivial. If we decide to keep a 1 second default here, we should at least consider keeping [this commit](https://github.com/sirugh/scoutfile/commit/6cdf31fe97692e3edfe12695c469e8e4e83bb7df)

**Comment by reason-bv** [comment]
Ok, fair arguments for the longer timeout.

File: docs/modules.md [hunk 1]
```diff
@@ -231,7 +231,11 @@ attributes on the created `<link>` element.
 - **injectionNode**: a DOM node into which the `<link>` element should be
 placed. The default insertion point is after the first `<script>` element on the page.
 - **timeout**: the time in milliseconds before the loading should be
-considered failed. Defaults to 1000ms.
+considered failed. Defaults to 10000ms.
+
+### loader timeout
+
+The loader module provides an extensible timeout that will invoke the provided callback with an error. In certain remote testing environments, the default timeout may not be sufficient for an initial page load. If you experience timeout errors when using the loader module you should experiment with upping this timeout using the provided `option.timeout`.
```
Comment (reason-bv) [inline docs/modules.md]: s/upping/raising/

**Comment by reason-bv** [comment]
The added docs are a start, but too buried.

They are in the right place, but not called out to the degree it should be. I think the main README.md needs the addition of a section on usage environments and caveats specifically to hold this issue on timeouts, which can then point to the docs on the loader as to where to get more information. The README.md section should also provide a code example for changing the timeout.

**Comment by sirugh** [comment]
I disagree that the README should hold reference to specific module implementation details. I think it belongs in the modules.md file or in a new loader doc under the docs/ folder.

File: README.md [hunk 2]
```diff
@@ -435,6 +435,14 @@ grunt.initConfig({
 });
 ```
 
+## Troubleshooting
+
+### loader timeout
+
+Understanding the loader module is core to usage of this repository. The default timeout provided may not be enough for environments where load times will be much slower than anticipated (ie. mobile, dial-up, etc). Certain browsers may also further obfuscate the issue by making it difficult to debug.
```
Comment (reason-bv) [inline README.md]: Understanding the loader module is critical to the usage of this repository. The default timeout is sufficient for common usage, but may not be large enough for environments where load times will be much slower than anticipated, such as slow mobile networks, dial-up connection, use of tunneling Selenium test services such as BrowserStack and Sauce Labs, and so forth. Older browsers such as IE8 may obfuscate timeout problems with other errors, making it difficult to debug such issues.

If you experience any issues with application load timeouts, review the documentation on the timeout option and experiment with larger timeout values.

File: README.md [hunk 3]
```diff
@@ -435,6 +435,14 @@ grunt.initConfig({
...
+
+For this reason, if you experience any issues with your application timing out after using this repository, check out the documentation on the [timeout option](https://github.com/bazaarvoice/scoutfile/blob/master/docs/modules.md#loader-timeout).
```
Comment (reason-bv) [inline README.md]: Move the link to footer link:

```
[timeout option][timeout-option]
```
```
[timeout-option]: ./docs/modules.md#loader-timeout
```

**Comment by aterranova-bv** [comment]
Confirmed that the code change accomplishes its goals.

But because I had to manually test that, can we add unit tests for this?  Specifically I'm looking at [this test](https://github.com/bazaarvoice/scoutfile/blob/master/test/unit/lib/browser/loader.js#L134).

Also, the default value in [this comment](https://github.com/bazaarvoice/scoutfile/blob/master/lib/browser/loader.js#L65) will need to change to the new value.

And finally, just to play devil's advocate, 10 seconds seems like an eternity.  Can we increase the value, but not quite as much?  Say, three (3) seconds?

**Comment by sirugh** [comment]
@aterranova-bv I think after 10 seconds we can guess that the resource actually DID time out and it's not taking longer to load because of bandwidth.

**Comment by sirugh** [comment]
Also, @aterranova-bv, getting between the loadScript invocation and the virtual dom actually loading the file would be pretty difficult. You'd have to somehow delay what the html script tag does BEFORE it loads the script but AFTER it has injected the script tag in the page.
